src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -370,12 +402,42 @@ private void AddHash()
                 // We can exclude more in the future.
 
                 StringBuilder.Append("/* HASH ");
-                Parameters.AppendHash(StringBuilder);
-                Parameters.AppendHashedParameterNames(StringBuilder);
+                if (forSmartV2Include)
+                {
+                    Parameters.AppendSmartScopeHash(StringBuilder);
+                    Parameters.AppendSmartScopeParameterNames(StringBuilder);
+                }
+                else
+                {
+                    Parameters.AppendHash(StringBuilder);
+                    Parameters.AppendHashedParameterNames(StringBuilder);
+                }
+
                 StringBuilder.AppendLine(" */");
             }
         }
 
+        /// <summary>
+        /// Marks parameters that were added after a specific point in time as SMART scope parameters.
+        /// </summary>
+        /// <param name="parametersBefore">The set of parameters that existed before the operation.</param>
+        /// <returns>List of new parameters that were added and marked as SMART scope parameters.</returns>
+        private List<SqlParameter> MarkNewParametersAsSmartScopeParameter(HashSet<SqlParameter> parametersBefore)
+        {
+            var parametersAfter = new HashSet<SqlParameter>(Parameters.ParametersToHash);
```

Comment:
Did we have something in the expression which already identified parameters are being part of the scope?

Reply:
We are passing the list of parameters to hash which we saved before applying the Smart v2 scope union expression. We have access to all the parameters to hash and we selected only the newly added parameters on the next line

Reply:
I understand the need to mark the expressions, and agree.